extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -264,6 +254,37 @@ class QueueProcessor {
         });
     }
 
+    _setupClients(sourceRole, targetRole) {
```

Comment:
I would rather decouple the auth managers from the `QueueProcessor` object, because we may want to keep the possibility to parallelize synchronizations later, which is not possible if cred managers are stored in the main queue processor object. Could be done by embedding all necessary info in a `ReplicationContext` object for example.